storage/src/adb/any.rs

```diff
@@ -195,35 +217,37 @@ impl<E: RStorage + Clock + Metrics, K: Array, V: Array, H: CHasher> Any<E, K, V,
         Ok(db)
     }
 
-    /// Update the location of `key` to `new_loc` in the snapshot, or insert it if the key isn't
-    /// already present.  If a `value` is provided, then it is used to see if the key is already
-    /// assigned that value, in which case this is a no-op, and `false` is returned.
+    /// Update the location of `key` to `new_loc` in the snapshot and return its old location, or
+    /// insert it if the key isn't already present. If a `value` is provided, then it is used to see
+    /// if the key is already assigned that value, in which case this is a no-op, and `None` is
+    /// returned.
     async fn update_loc(
         snapshot: &mut Index<EightCap, u64>,
         log: &mut Journal<E, Operation<K, V>>,
         key: K,
         value: Option<&V>,
         new_loc: u64,
-    ) -> Result<bool, Error> {
+    ) -> Result<Option<Option<u64>>, Error> {
```

Comment:
I'd recommend structuring this "complex" return value as a `struct` or `enum` rather than this wrapping `Option`?

Maybe something like `Enum { Skipped, Added, Updated(u64)}`?

Reply:
Not a huge fan of ever having to write `Some(None)`